lib/launchers/o_c_m_cluster.rb

```diff
@@ -0,0 +1,295 @@
+    end
+
+    private :default_opts
+
+    def to_seconds(string)
+      regex_m = /^(\d+)\s*(m|min|minutes|mins)+$/
+      regex_h = /^(\d+)\s*(h|hour|hours|hrs)+$/
+      regex_d = /^(\d+)\s*(d|day|days)+$/
+      regex_w = /^(\d+)\s*(w|week|weeks|wks)+$/
+      case string
+      when regex_m
+        return string.match(regex_m)[1].to_i * 60
+      when regex_h
+        return string.match(regex_h)[1].to_i * 60 * 60
+      when regex_d
+        return string.match(regex_d)[1].to_i * 24 * 60 * 60
+      when regex_w
+        return string.match(regex_w)[1].to_i * 7 * 24 * 60 * 60
+      else
+        raise "Cannot convert '#{string}' to seconds!"
+      end
+    end
+
+    # Generate a cluster data used for creating OSD cluster
+    def generate_cluster_data(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @num_nodes
+        json_data.merge!({"nodes" => {"compute" => @num_nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @aws_account_id && @aws_access_key && @aws_secret_key
+        json_data.merge!({"aws" => {"account_id":@aws_account_id, "access_key_id":@aws_access_key, "secret_access_key":@aws_secret_key}})
+        json_data.merge!({"byoc" => true})
+      end
+
+      return json_data
+    end
+
+    def download_ocm_cli
+      url = ENV['OCM_CLI_URL']
+      unless url
+        url_prefix = ENV['OCM_CLI_URL_PREFIX'] || 'https://github.com/openshift-online/ocm-cli/releases/download/v0.1.46'
+        if OS.mac?
+          url = "#{url_prefix}/ocm-darwin-amd64"
+        elsif OS.linux?
+          url = "#{url_prefix}/ocm-linux-amd64"
+        else
+          raise "Unsupported OS"
+        end
+      end
+      ocm_path = File.join(Host.localhost.workdir, 'ocm')
+      File.open(ocm_path, 'wb') do |file|
+        @result = Http.get(url: url, raise_on_error: true) do |chunk|
+          file.write chunk
+        end
+      end
+      File.chmod(0775, ocm_path)
+      return ocm_path
+    end
+
+    def shell(cmd, output = nil)
+      if output
+        res = Host.localhost.exec(cmd, single: true, stderr: :stdout, stdout: output, timeout: 3600)
+      else
+        res = Host.localhost.exec(cmd, single: true, timeout: 3600)
+      end
+      if res[:success]
+        return res[:response]
+      else
+        raise "Error when executing '#{cmd}'. Response: #{res[:response]}"
+      end
+    end
+
+    def exec(cmd)
+      unless @ocm_cli
+        @ocm_cli = download_ocm_cli
```

Comment:
btw might be a good idea to first check `ocm` binary in local PATH and only then download

Reply:
Yes, I was thinking also about that as well. There are more such enhancements which could be done but they can be implemented later.